Front page improvements (#413)
packit/packit.dev: content/_index.md

disableToc: false
---

{{< columns >}}

### What is Packit?

1. a [CLI tool]({{< ref "cli" >}}), which you can
   [install locally](/docs/guide/#have-packit-tooling-installed-locally)
   and start using right away,

2. a [service]({{< ref "packit-service" >}}), which provides you feedback

>    on how is your project integrated with [Fedora Linux](https://getfedora.org) or [CentOS Stream](https://www.centos.org/centos-stream/).

Reviewer: ```suggestion
   on how is your project integrated with [Fedora Linux](https://getfedora.org), [CentOS Stream](https://www.centos.org/centos-stream/) and other distributions.
```

Copr supports much more :)

   <--->

### Who should use Packit?

> Any upstream maintainer who would like to automate validation, testing or releasing of code changes
> in his project(s).

Reviewer: ```suggestion
Upstream maintainers who would like to automate validation, testing and releasing of code changes
in downstream distributions.
```
I mainly wanted to make this gender-neutral.

{{< /columns >}}

### What key features does Packit offer?

{{< columns >}}

#### Building RPMs

Packit uses [Copr build system](https://copr.fedorainfracloud.org/) to build RPMs from your code changes,
so you can validate your package will be correctly built from each pull request or have automation around your Copr
repository containing builds from your upstream releases/commits.
<--->

#### Testing

Packit runs your tests in [Testing Farm](https://docs.testing-farm.io/general/0.1/index.html) and can test either the RPM package built in Copr or
trigger tests directly on different platforms.
<--->

#### Automating Fedora releases

With Packit, you can easily

> get the new upstream releases to the [Fedora dist-git](https://src.fedoraproject.org), automatically submit builds in [Koji](https://koji.fedoraproject.org/koji/), and we are currently

Reviewer: ```suggestion
get your upstream releases to the [Fedora dist-git](https://src.fedoraproject.org), automatically submit builds in [Koji](https://koji.fedoraproject.org/koji/), and we are currently
```

> finalizing implementation for automatic creation of [Bodhi](https://bodhi.fedoraproject.org) updates!

Author: :+1: will reword

{{< /columns >}}

### How does Packit look like in action?

![Packit workflow](/images/build_test_workflow.jpg)

### Why start using Packit?

1. _Easy to configure_ - Feel free to jump into [our guide]({{< ref "guide" >}}) or check directly the [configuration documentation](/docs/configuration).

> 2. _Stable users_ - Systemd, cockpit, avocado, ansible-lint and many other projects. If you are interested,

Reviewer: could you please make these links to the github projects?

disableToc: false
---

{{< columns >}}

### What is Packit?

1. a [CLI tool]({{< ref "cli" >}}), which you can
   [install locally](/docs/guide/#have-packit-tooling-installed-locally)
   and start using right away,

2. a [service]({{< ref "packit-service" >}}), which provides you feedback
   on how is your project integrated with [Fedora Linux](https://getfedora.org), [CentOS Stream](https://www.centos.org/centos-stream/)
   and other distributions.
   <--->

### Who should use Packit?

Upstream maintainers who would like to automate validation, testing and releasing of code changes

> in downstream distributions.

Reviewer: sounds good, I would reword it to:

"Upstream projects that want to build RPM packages out of pull requests, commits or releases and bring those releases to Fedora Linux."

> in downstream distributions.

Author: Thanks for the suggestions! I agree with mentioning building RPM packages and changing upstream maintainers to upstream projects.
> "Upstream projects that want to build RPM packages out of pull requests, commits or releases and bring those releases to Fedora Linux."

This is nice and clear, but I personally am missing the testing part, so I would adjust it a little bit, WDYT?

{{< /columns >}}

### What key features does Packit offer?

{{< columns >}}

#### Building RPMs

Packit uses [Copr build system](https://copr.fedorainfracloud.org/) to build RPMs from your code changes,
so you can validate your package will be correctly built from each pull request or have automation around your Copr

> repository containing builds from your upstream releases/commits.

Author: Yeah, I struggled with this sentence :D I will try to reword it considering your suggestion.

<--->

#### Testing

Packit runs your tests in [Testing Farm](https://docs.testing-farm.io/general/0.1/index.html) and can test either the RPM package built in Copr or
trigger tests directly on different platforms.
<--->

#### Automating Fedora releases

With Packit, you can easily

> get your upstream releases to the [Fedora dist-git](https://src.fedoraproject.org), automatically submit builds in [Koji](https://koji.fedoraproject.org/koji/), and we are currently

Reviewer 2: Do we have this actually somewhere documented (from top view)? I am thinking of link "learn more".

Reviewer 3: The "get your upstream releases" is documented as `propose-downstream` job and `Koji` builds will be available once we merge #403. (We can add the link later on.)

And the whole overview can be seen here #404. (We are discussing [here](https://github.com/packit/packit.dev/pull/404#discussion_r826630151) about reusing/updating this to be the overall guide or creating a new, similar, text that will cover this.

Author: I would probably add the link to the whole overview when it will be merged (not sure about linking the specific jobs, since they should be able to reach from the overview and the previous `features` also do not directly link the jobs.

finalizing implementation for automatic creation of [Bodhi](https://bodhi.fedoraproject.org) updates!
{{< /columns >}}

### How does Packit look like in action?

![Packit workflow](/images/build_test_workflow.jpg)

### Why start using Packit?

1. _Easy to configure_ - Feel free to jump into [our guide]({{< ref "guide" >}}) or check directly the [configuration documentation](/docs/configuration).

> 2. _Stable users_ - [Systemd](https://github.com/systemd/systemd/), [cockpit](https://github.com/cockpit-project/cockpit), [avocado](https://github.com/avocado-framework/avocado), [ansible-lint](https://github.com/ansible-community/ansible-lint) and many other projects. If you are interested,

Reviewer 2: Hmm, should we link the main page of these projects - or should we link directly to packit.yaml in their git repo?

Reviewer: I was also thinking about linking to packit.yaml but... we'd need to make sure they don't rename it or move around (or even drop 😩), so the conclusion in my head was to link to the github project and let people "discover" how they use packit (open a PR, click on checks, look at packit.yaml)

Author: We could still use permalinks to the current versions of their packit configs, but that could get outdated as well, so I am also not sure :confused:

Reviewer: maybe it's finally time to start checking URLs in our docs if they are up to date :D

anyway, I'm +1 to link directly the configs (since that may be the first thing people look for)

Reviewer 2: > We could still use permalinks to the current versions of their packit configs, but that could get outdated as well, so I am also not sure confused

We can link to specific treeish. E.g.,  https://github.com/systemd/systemd/blob/d15e1a29e3aab04ee79d5e3ec8e1e65fca78e165/.packit.yml But this, on the other hand, does not get updates when we update some of our API (format of packit.yaml), which will be incompatible with the current format. Hmm, but that may or may not happen. Definitely in the distant future. :)

Author: > We can link to specific treeish. E.g., https://github.com/systemd/systemd/blob/d15e1a29e3aab04ee79d5e3ec8e1e65fca78e165/.packit.yml 

Yes, that's what I meant by permalinks.

Agreed, changed it to the links to configs.
